test/cpp_extensions/libtorch_agnostic_extension/test/test_libtorch_agnostic.py

```diff
@@ -222,11 +222,52 @@ def test_fill_infinity(self, device):
         def test_default_constructor(self):
             import libtorch_agnostic
 
-            is_defined_defined = libtorch_agnostic.ops.test_default_constructor(True)
-            self.assertTrue(is_defined_defined)
+            defined_tensor_is_defined = libtorch_agnostic.ops.test_default_constructor(
+                True
+            )
+            self.assertTrue(defined_tensor_is_defined)
+
+            undefined_tensor_is_defined = (
+                libtorch_agnostic.ops.test_default_constructor(False)
+            )
+            self.assertFalse(undefined_tensor_is_defined)
+
+        @onlyCPU
+        def test_default_constructor_memory_cleanup(self):
+            import subprocess
+            import sys
+
+            import libtorch_agnostic
+
+            def run_in_subprocess():
+                import gc
+                import os
+
+                import psutil
+
+                process = psutil.Process(os.getpid())
+                gc.collect()
+
+                initial_memory = process.memory_info().rss
+
+                for _ in range(100):
+                    libtorch_agnostic.ops.test_default_constructor(False)
+
+                final_memory = process.memory_info().rss
+                memory_increase = final_memory - initial_memory
```

Comment:
woah a way to test CPU memory! how confident are you that this test is testing the right thing? This looks okay to me but I am not an expert

Reply:
I tried creating a list of tensors an the RSS increased, so I felt somewhat confident this was doing the right thing

cc @albanD do you know :P

Reply:
This will most likely be flaky as we don't control what else might grab cpu memory.
I would recommend running this test on cuda and check for cuda memory which is very reliable for this kind of things.
Don't forget to add @serialTest() decorator to avoid cross polution of course ;)